### PR TITLE
[browser] refactor entropy

### LINKED ISSUE #93398: [wasm] MT browser-bench crash
I was trying to measure wasm/MT perf recently. The browser-bench is crashing for me, during the app start measurements, in this place:

```
AppStart, Reach managed count: 29, per call: 170.79310344827587ms, total: 4.953s
AppStart, Reach managed count: 31, per call: 178.83870967741936ms, total: 5.544s
dotnet.native.js:4738 Uncaught TypeError: Failed to execute 'getRandomValues' on 'Crypto': The provided ArrayBufferView value must not be shared.
    at Object.getBatchedRandomValues (dotnet.native.js:4738:11)
    at _dotnet_browser_entropy (dotnet.native.js:4749:17)
    at dotnet.native.wasm:0x2110c7
    at dotnet.native.wasm:0x210f5d
    at dotnet.native.wasm:0x21f50
    at dotnet.native.wasm:0x1ecb4
    at dotnet.native.wasm:0x12fdd
    at dotnet.native.wasm:0x200e3
    at dotnet.native.wasm:0xfa6b2
    at dotnet.native.wasm:0xc82b7
```

### PR DESCRIPTION
**Pull request by pavelsavara**
Fixes https://github.com/dotnet/runtime/issues/93398

- the real fix is addressing know issue 
    - "In some cases, `instanceof SharedArrayBuffer` returns false even though buffer is an SAB."
- I did it by refactoring away `pal_random.lib.js` and including the logic into the rest of the TypeScript code we have
    - because we already have existing solution for it

### CODE REVIEW

File: src/mono/wasm/runtime/crypto.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    // When threading is enabled, Chrome doesn't want SharedArrayBuffer to be passed to crypto APIs
+    const needsCopy = isSharedArrayBuffer(memoryView.buffer);
+    const targetBuffer = needsCopy
+        ? new Uint8Array(bufferLength)
+        : targetView;
+
+    // fill the targetBuffer in batches of batchedQuotaMax
+    for (let i = 0; i < bufferLength; i += batchedQuotaMax) {
+        const targetBatch = new Uint8Array(targetBuffer, i, Math.min(bufferLength - i, batchedQuotaMax));
+        crypto.getRandomValues(targetBatch);
```
Comment (radekdoulik) [inline src/mono/wasm/runtime/crypto.ts:21]: the original implementation was detecting missing crypto in the browser. do we drop that intentionally? I think it makes sense, the coverage is >96% https://caniuse.com/cryptography

**Review by radekdoulik — APPROVED** [review]

File: src/mono/wasm/runtime/crypto.ts (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/crypto.ts]: I added back the original logic. Thanks!

**Comment by radekdoulik** [comment]
the debugger failure is https://github.com/dotnet/runtime/issues/89409

**Pull request merged**